packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -370,6 +370,10 @@ class SliverChildBuilderDelegate extends SliverChildDelegate {
   /// Will be called only for indices greater than or equal to zero and less
   /// than [childCount] (if [childCount] is non-null).
   ///
+  /// May result in an infinite loop or run out of memory if [childCount] is null
+  /// and the [builder] always provide a zero-size widget, such as `Container()`
+  /// or `SizedBox.shrink()`.
```

Comment:
Is there a way we can catch this and throw an error?

Reply:
I've tried both solutions, but neither is perfect.
1. Timeout check. When an exception is thrown when time out(about 3 seconds), the Flutter error thread will take about ten seconds(or more) to collect error information (because thousands of children have been created at this time), and the final error message printed to the console will overflow, and the user cannot well informed of what happened;
2. When 100 children of zero size are created in a row, throw an exception, which is a bit hacky and maybe misjudged.
Wondering if you have any suggestions?

Reply:
Hmm.. you are right, this is very difficult to catch accurately.
We cannot add timeouts (https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#never-check-if-a-port-is-available-before-using-it-never-add-timeouts-and-other-race-conditions), and it is possible the user could have zero sized children in the thousands preceding a child that actually has a size.

Reply:
Perhaps we can word it a bit differently, right now it sounds like a warning against this scenario. What should the user do then if they run into this?

Set a child count, return null from builder, or provide children with a size sound like options, what do you think?